apollo-router/src/plugins/telemetry/config_new/mod.rs

```diff
@@ -28,17 +29,31 @@ pub(crate) mod spans;
 pub(crate) trait Selectors {
     type Request;
     type Response;
+    type EventResponse;
+
     fn on_request(&self, request: &Self::Request) -> Vec<KeyValue>;
     fn on_response(&self, response: &Self::Response) -> Vec<KeyValue>;
+    fn on_event_response(&self, _response: &Self::EventResponse, _ctx: &Context) -> Vec<KeyValue> {
```

Comment:
should this be `on_response_event`? oh maybe it refers to defer and subscription events only? is there a design doc somewhere?

Reply:
It refers to event coming in the supergraph stream so yes mainly subscription and defer events. Maybe your suggestion is still valid. WDYT @BrynCooke ?

Reply:
Fixed with `on_response_event` thanks for the suggestion